feat: 인증된 사용자 정보를 조회하는 유틸 클래스(CurrentMemberUtil) 추가

higukang opened the pull request:
# 🚀 Pull Request

**[작업 내용을 간략히 적어주세요]**
인증된 사용자 정보를 조회기능 추가


## #️⃣ 연관된 이슈

#233 

## 📋 작업 내용
- SecurityContextHolder를 활용해 현재 인증된 사용자의 이메일 조회 메서드 추가
- 인증되지 않은 사용자에 대한 예외 처리 로직 추가
- 테스트용 코드로 관리자 계정(admin@example.com) 반환 임시 적용 (TODO 주석 추가)

higukang committed:
feat: 인증된 사용자 정보를 조회하는 유틸 클래스(CurrentMemberUtil) 추가

- SecurityContextHolder를 활용해 현재 인증된 사용자의 이메일 조회 메서드 추가
- 인증되지 않은 사용자에 대한 예외 처리 로직 추가
- 테스트용 코드로 관리자 계정(admin@example.com) 반환 임시 적용 (TODO 주석 추가)

#Isue: 233

oowtl commented on module-service/src/main/java/com/checkping/service/member/util/CurrentMemberUtil.java:
```diff
@@ -0,0 +1,54 @@
+import java.util.Optional;
+
+
+@Component
```
이 유틸은 주입받아서 사용하는 것인가요?

oowtl commented on module-service/src/main/java/com/checkping/service/member/util/CurrentMemberUtil.java:
```diff
@@ -0,0 +1,54 @@
+     * @return 현재 인증된 사용자의 MemberResponseDto
+     * @throws MemberException 인증된 사용자를 찾을 수 없는 경우 예외 발생
+     */
+    public Member getCurrentMember() {
```
Member 가 필요하면 이 메서드를 사용하면 될까요?

jaseongkim replied:
네 그렇습니다.

higukang committed:
feat: JWT 토큰 저장 위치 변경

- AccessTokenNotFound 예외 추가
- RefreshTokenNotFoundException 예외 추가
- ResponseUtil : 엑세스 토큰을 쿠키에 저장하도록 변경
= ReissueController : 엑세스 토큰을 쿠키에 저장하도록 변경
- JWTFilter : 쿠키에서 엑세스 토큰을 찾아 검증하도록 수정
- ErrorCode : ACCESS_TOKEN_NOT_FOUND, REFRESH_TOKEN_NOT_FOUND 추가
- CustomLogoutFilter : 엑세스 토큰도 쿠키에서 삭제하도록 수정

#Issue: 241

higukang committed:
Merge branch 'feature/241-jwt-토큰-저장-위치-변경' into feature/233-인증된-사용자-정보를-조회-기능-추가
token 저장 위치 쿠키로 변경된 내용 적용

higukang committed:
feat: 인증된 사용자의 이메일 추출 로직 개선

- SecurityContextHolder에서 가져온 principal의 타입에 따라 이메일 추출 로직 추가
  - CustomUserDetails, UserDetails, String 타입별 처리
  - 예외 처리 추가: 예상하지 못한 principal 타입의 경우 MemberException 발생

# 개선 사항
- 인증 정보 로드 실패 시 구체적인 예외 발생

#Issue: 233

higukang committed:
feat: 현재 로그인된 사용자 정보를 반환하는 API 추가

- AuthService 클래스 생성:
  - CurrentMemberUtil을 사용하여 현재 로그인된 사용자 정보(MemberResponseDto) 반환
- AuthController 클래스 생성:
  - `/auth/me` 엔드포인트에서 현재 로그인된 사용자 정보 제공
  - AuthService와 연동

- 현재 인증된 사용자 정보 확인 기능 추가

#Issue: 233

higukang committed:
chore: 인증 컨트롤러 경로 수정
